components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -1494,4 +1411,454 @@ private static Validator resourceToValidatorObject(
+                    return saveUpdatedResource(resource, issuerDNMap);
+                }
+            }
+            return null;
+        } catch (IOException | ConfigurationManagementException e) {
+            throw new CertificateValidationException("Error while processing the resource", e);
+        }
+    }
+
+    /**
+     * Delete the ca certificate from the certificate manager.
+     *
+     * @param certificateId Certificate id
+     * @param tenantDomain Tenant domain
+     * @throws CertificateValidationException Error when deleting certificate
+     */
+    public static void deleteCACertificateFromCertificateManager(String certificateId,
+                                                                 String tenantDomain)
+            throws CertificateValidationException {
+
+        try {
+            CertValidationDataHolder.getInstance()
+                    .getCertificateManagementService()
+                    .deleteCertificate(certificateId, tenantDomain);
+        } catch (CertificateMgtException e) {
+            log.debug("Error when getting certificate from certificate manager.", e);
+            if (CertificateMgtErrors.ERROR_CERTIFICATE_DOES_NOT_EXIST.getCode().equals(e.getErrorCode())) {
+                throw new CertificateValidationException
+                        ("Certificate with the id: " + certificateId + " does not exist.", e);
+            } else {
+                throw new CertificateValidationException("Error when getting certificate from certificate manager.", e);
+            }
+        }
+    }
+
+    private static CertObject processCertificateInStore(String tenantDomain, X509Certificate certificate,
+                                                        String certificateId, boolean isUpdate)
+            throws CertificateValidationException {
+
+        try {
+            org.wso2.carbon.identity.configuration.mgt.core.model.Resource resource = CertValidationDataHolder
+                    .getInstance().getConfigurationManager()
+                    .getResourceByTenantId(IdentityTenantUtil.getTenantId(tenantDomain),
+                            X509_CA_CERT_RESOURCE_TYPE, CERTS);
+            InputStream inputStream = getResourceFileInputStream(resource);
+            if (inputStream == null) {
+                return null;
+            }
+
+            IssuerDNMap issuerDNMap = ModelSerializer.deserializeIssuerDNMap(convertInputStreamToString(inputStream));
+            String issuerDN = getNormalizedName(certificate.getIssuerDN().getName());
+            String serialNumber = getNormalizedName(certificate.getSerialNumber().toString());
+
+            List<String> ocspUrls = getValidationUrls(certificate, tenantDomain, OCSP_VALIDATOR);
+            List<String> crlUrls = getValidationUrls(certificate, tenantDomain, CRL_VALIDATOR);
+
+            CertObject certObject = new CertObject();
+            certObject.setCertId(isUpdate ? certificateId : UUID.randomUUID().toString());
```

Comment:
Why we need to generate an UUID?

Reply:
fixed with https://github.com/wso2-extensions/identity-x509-commons/pull/57/commits/52f92488d265780c25c0d23c5ec638ca00bd3f59